cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py

```diff
@@ -74,3 +76,22 @@ def get_repeated_sample_names(case: TomteCase) -> list[str]:
 def get_repeated_sample_name_errors(case: TomteCase) -> list[RepeatedSampleNameError]:
     sample_names = get_repeated_sample_names(case)
     return [RepeatedSampleNameError(sample_name=name, case_name=case.name) for name in sample_names]
+
+
+def get_father_sex_errors(case: TomteCase) -> list[InvalidFatherSexError]:
+    errors = []
+    children: list[TomteSample] = case.get_samples_with_father()
+    for child in children:
+        if is_father_sex_invalid(child=child, case=case):
+            error = create_father_sex_error(case=case, sample=child)
+            errors.append(error)
+    return errors
+
+
+def is_father_sex_invalid(child: TomteSample, case: TomteCase) -> bool:
+    father: TomteSample | None = case.get_sample(child.father)
+    return father and father.sex != Sex.MALE
```

Comment:
This makes the checks not atomic. If the father does not exist in the case we would not want a InvalidFatherSexError. That does not seem like what we want

Reply:
That won't result in an error here, if the father does not exist the check returns false and no error is created 👍

Reply:
But maybe I should add a test for it 🤔